Lib/concurrent/futures/thread.py

```diff
@@ -148,4 +150,5 @@ def shutdown(self, wait=True):
         if wait:
             for t in self._threads:
                 t.join()
+        self._threads.clear()
```

Comment:
I'm working on race conditions since 3-6 months, and I saw so many non deterministic behaviours. I would like to make Python test suite more reliable, more reproductible.